restler/utils/logger.py

```diff
@@ -434,28 +436,30 @@ def add_hash(replay_filename):
         print("-------------", file=log_file)
         for bucket_class in bug_buckets:
             for seq_hash in bug_buckets[bucket_class]:
-                # seq_obj = (sequence definition, bug reproduced boolean)
-                seq_obj = bug_buckets[bucket_class][seq_hash]
-                name_header = bucket_class
+                bug_bucket = bug_buckets[bucket_class][seq_hash]
                 bucket_hash = f"{seq_hash}_{bucket_class}"
+                name_header = bucket_class
                 if bucket_hash not in Bugs_Logged:
                     try:
                         filename = log_new_bug()
-                        Bugs_Logged[bucket_hash] = (filename, bug_hash)
+                        Bugs_Logged[bucket_hash] = BugTuple(filename, bug_hash, bug_bucket.retry_count, bug_bucket.reproduce_count)
                         add_hash(filename)
                     except Exception as error:
                         write_to_main(f"Failed to write bug bucket log: {error!s}")
                         filename = 'Failed to create replay log.'
                 else:
-                    filename = Bugs_Logged[bucket_hash][0]
-                if seq_obj[1]:
-                    name_header = f'{name_header} - Bug was reproduced - {filename}'
+                    filename = Bugs_Logged[bucket_hash].filename_of_replay_log
+                    this_bug_hash = Bugs_Logged[bucket_hash].bug_hash
+                    Bugs_Logged[bucket_hash] = BugTuple(filename, this_bug_hash, bug_bucket.retry_count, bug_bucket.reproduce_count)
+
+                if bug_bucket.reproducible:
+                    print(f'{name_header} - Bug was reproduced - {filename}', file=log_file)
                 else:
-                    name_header = f'{name_header} - Unable to reproduce bug - {filename}'
-                print(name_header, file=log_file)
-                print(f"Hash: {Bugs_Logged[bucket_hash][1]}", file=log_file)
-                sequence = seq_obj[0]
-                for request in sequence:
+                    print(f'{name_header} - Unable to reproduce bug - {filename}', file=log_file)
+                    print(f'Attempted to reproduce {Bugs_Logged[bucket_hash].reproduce_attempts} additional time(s); '
```

Comment:
>additional [](start = 97, length = 10)

Is 'additional' needed?

Reply:
I added that because it looked a little weird and could be confusing, since the count only refers to attempts after the initial bug is logged.  For instance, if only the initial attempt fails and then we retry two more times it would say, "unable to reproduce bug", but then on the next line, "Attempted to reproduce 2 times; Reproduced 2 times."

If this looks ok I can get rid of it - or if all attempts (including the first attempt) should be part of the count, I can make that change, too.

Reply:
By definition, if the previous line says "unable to reproduce bug", then number_of_attempts > number_of_repros . Is this correct? 
If so, "additional" can be removed. OK?

Reply:
@patgod that's sort of not correct. If the first attempt was "unable to reproduce bug", but every attempt to reproduce the bug after that succeeds number_of_attempts == number_of_repros.  I can add the initial reproduce attempt that fails to the number_of_attempts count and then get rid of additional - probably makes the most sense

Reply:
@rifiles yes, the latter makes more sense (so that we always have number_of_attempts > number_of_repros after we print "unable to reproduce bugs"). Thanks!